niceman/distributions/tests/test_conda.py

```diff
@@ -8,45 +8,56 @@
 # ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
 import collections
 import os
+import pytest
 
 import sys
 from appdirs import AppDirs
 from subprocess import call
+from unittest import SkipTest
 
 import yaml
 import attr
 from niceman.formats.niceman import NicemanProvenance
-from niceman.tests.utils import skip_if_no_network
-
-try:
-    import apt
-except ImportError:
-    apt = None
+from niceman.tests.utils import skip_if_no_network, assert_is_subset_dict_recur
 
 import json
 
 from niceman.distributions.conda import CondaTracer
 
 
-def create_test_conda(test_dir):
+@pytest.fixture
+def get_conda_test_dir():
+    # Return none if no network is avaialble
+    if os.environ.get('NICEMAN_TESTS_NONETWORK'):
+        return None
+    dirs = AppDirs('niceman')
+    test_dir = os.path.join(dirs.user_cache_dir, 'conda_test')
     if os.path.exists(test_dir):
-        return
+        return test_dir
+    # Miniconda isn't installed, so install it
+    if sys.platform.startswith('darwin'):
+        miniconda_sh = "Miniconda2-latest-MacOSX-x86_64.sh"
+    elif sys.platform.startswith('linux'):
+        miniconda_sh = "Miniconda2-latest-Linux-x86_64.sh"
+    else:
+        raise ValueError("Conda test not supported with platform %s " %
+                         sys.platform)
     call("mkdir -p " + test_dir + "; "
          "cd " + test_dir + "; "
-         "curl -O https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh; "
-         "bash -b Miniconda2-latest-Linux-x86_64.sh -b -p ./miniconda; "
-         "./miniconda/bin/conda create -y -n mytest python=2.7;" + \
-         "./miniconda/envs/mytest/bin/conda install -y xz -n mytest;" + \
+         "curl -O https://repo.continuum.io/miniconda/" + miniconda_sh + "; "
+         "bash -b " + miniconda_sh + " -b -p ./miniconda; "
+         "./miniconda/bin/conda create -y -n mytest python=2.7; "
+         "./miniconda/envs/mytest/bin/conda install -y xz -n mytest; "
          "./miniconda/envs/mytest/bin/pip install rpaths;",
          shell=True)
+    return test_dir
 
 
-@skip_if_no_network
-def test_conda_manager_identify_distributions():
-    dirs = AppDirs('niceman')
-    test_dir = os.path.join(dirs.user_cache_dir, 'conda_test')
-    create_test_conda(test_dir)
-    print (test_dir + "\n")
+def test_conda_manager_identify_distributions(get_conda_test_dir):
+    # Skip if network is not available (skip_if_no_network fails with fixtures)
+    test_dir = get_conda_test_dir
+    if not test_dir:
+        raise SkipTest("Skipping since no network settings")
```

Comment:
ha, apparently `skip_if_no_network` is smart enough so you could just use it as a function inside (just call it within the test and it would raise SkipTest if no network setting)

Reply:
Done 2e1320f